lib/ansible/galaxy/login.py

```diff
@@ -80,8 +81,14 @@ def remove_github_token(self):
         retrieve the token after creation, so we are forced to create a new one.
         '''
         try:
-            tokens = json.load(open_url(self.GITHUB_AUTH, url_username=self.github_username,
-                               url_password=self.github_password, force_basic_auth=True,))
+            opened = open_url(self.GITHUB_AUTH,
+                              url_username=self.github_username,
+                              url_password=self.github_password,
+                              force_basic_auth=True,)
+            resp = opened.read()
+            if sys.version_info.major > 2:
```

Comment:
Six is pretty easy to use here:
 `   if not six.PY2:`

Note: Don't use `if six.PY3` because there might one day be a `six.PY4`

Reply:
Cool!  I've never really used it.  I used the greater than ">" when checking major version in case there would ever be a python4.

Reply:
I went to look at the unit test.  It looks like six.PY3 is being imported but not actually used in the test.  It seems that the only test that is run on "login" is that the option parser correctly parsed the command.  I would like to see where a unit test already exists to to perform a successful login in python2.  I just looked in the ```test/units/cli``` directory, since that seemed to be the most likely place it would be.

Reply:
```
tokens = json.loads(to_text(resp.read(), errors='surrogate_or_strict'))
```
no need for checking Python version. `to_text` will take care of that.

Reply:
where is ```to_text```?

Reply:
https://github.com/ansible/ansible/blob/88a738b0bacd32d0027f71682d7965aad3c9a040/lib/ansible/module_utils/_text.py#L169

Reply:
I tried ``to_text``

diff:
```diff --git a/lib/ansible/galaxy/login.py b/lib/ansible/galaxy/login.py
index 6fa221a2e8..775ba8c559 100644
--- a/lib/ansible/galaxy/login.py
+++ b/lib/ansible/galaxy/login.py
@@ -31,6 +31,7 @@ from ansible.module_utils.six.moves import input
 from ansible.module_utils.six.moves.urllib.parse import quote as urlquote, urlparse
 from ansible.module_utils.six.moves.urllib.error import HTTPError
 from ansible.module_utils.urls import open_url
+from ansible.module_utils._text import to_text
 from ansible.utils.color import stringc
 
 try:
@@ -86,9 +87,8 @@ class GalaxyLogin(object):
                               url_password=self.github_password,
                               force_basic_auth=True,)
             resp = opened.read()
-            if sys.version_info.major > 2:
-                resp = resp.decode()
-            tokens = json.loads(resp)
+            tokens = json.loads(to_text(opened.read(),
+                                        errors='surrogate_or_strict'))
         except HTTPError as e:
             res = json.load(e)
             raise AnsibleError(res['message'])
@@ -114,10 +114,8 @@ class GalaxyLogin(object):
                               url_username=self.github_username,
                               url_password=self.github_password,
                               force_basic_auth=True, data=args)
-            resp = opened.read()
-            if sys.version_info.major > 2:
-                resp = resp.decode()
-            data = json.loads(resp)
+            data = json.loads(to_text(opened.read(),
+                                      errors="surrogate_or_strict"))
         except HTTPError as e:
             res = json.load(e)
             raise AnsibleError(res['message'])
```

traceback:
```(testansible) umeboshi@bard:~/workspace/A/ansible$ ansible-galaxy login -vvv
ansible-galaxy 2.7.0.dev0 (galaxy-login-py3 ae43eca45a) last updated 2018/08/06 14:54:52 (GMT -500)
  config file = None
  configured module search path = ['/freespace/home/umeboshi/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']                                                                                       
  ansible python module location = /freespace/home/umeboshi/workspace/A/ansible/lib/ansible
  executable location = /freespace/home/umeboshi/.virtualenvs/testansible/bin/ansible-galaxy
  python version = 3.5.3 (default, Jan 19 2017, 14:11:04) [GCC 6.3.0 20170118]
No config file found; using defaults
Opened /freespace/home/umeboshi/.ansible_galaxy


We need your Github login to identify you.
This information will not be sent to Galaxy, only to api.github.com.
The password will not be displayed.

Use --github-token if you do not want to enter your password.

Github Username: umeboshi2
Password for umeboshi2: 
ERROR! Unexpected Exception, this is probably a bug: Expecting value: line 1 column 1 (char 0)
the full traceback was:

Traceback (most recent call last):
  File "/freespace/home/umeboshi/workspace/A/ansible/bin/ansible-galaxy", line 118, in <module>
    exit_code = cli.run()
  File "/freespace/home/umeboshi/workspace/A/ansible/lib/ansible/cli/galaxy.py", line 162, in run
    self.execute()
  File "/freespace/home/umeboshi/workspace/A/ansible/lib/ansible/cli/__init__.py", line 155, in execute
    fn()
  File "/freespace/home/umeboshi/workspace/A/ansible/lib/ansible/cli/galaxy.py", line 570, in execute_login
    github_token = login.create_github_token()
  File "/freespace/home/umeboshi/workspace/A/ansible/lib/ansible/galaxy/login.py", line 110, in create_github_token
    self.remove_github_token()
  File "/freespace/home/umeboshi/workspace/A/ansible/lib/ansible/galaxy/login.py", line 91, in remove_github_token
    errors='surrogate_or_strict'))
  File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Thanks for the suggestions.